### DISCUSSION TITLE
How do you install Foundry permanently? New terminals return Command not found (Linux/Mac OS)

### DISCUSSION THREAD

**Discussion by gurundoro** [Cyfrin/foundry-full-course-cu#2389] [Q&A]
Everytime I open a new terminal and run any Foundry commands, I get a command not found error and have to reinstall it. How can I install it globally permanently?

**Comment by cosmostx7** [comment]
These commands are used to install foundry on a local machine:
`curl -L https://foundry.paradigm.xyz | bash`
`source ~/.bashrc `
`foundryup`

Are you using these or anything else?
What specific error you're facing?

**Reply by gurundoro** [reply]
This is how I've been installing it and it works. But my issue is that whenever I open a new terminal I have to reinstall it. 

Say for example I open a terminal to run anvil,  I'll get 'command not found' and then I have to install Foundry again in that terminal and then it works. I'd like to not have to do that.

**Comment by EngrPips** [comment]
Hello @gurundoro, Did you install Foundry with the command posted by @cosmostx7?

**Reply by gurundoro** [reply]
Hi @EngrPips. I did install Foundry with those commands and it works. The issue is that for every new terminal i open, e.g to run anvil,  I have to reinstall it otherwise I get command not found.

**Reply by EngrPips** [reply]
I installed it that way, and I didn't run into this problem. I am unsure what could be wrong.

**Reply by cosmostx7** [reply]
There might be some conflict with the system or variables.

**Comment by EngrPips** [comment]
Interesting.

**Comment by yue13-yang** [comment]
Had the same issue, foundry is not found every time I open a new terminal. 
There are two issues in my case. 
First, my default shell is zsh (for newer mac os, it's defaulted to zsh, but it's bash before)  
While installing foundry, it set the path to your default shell, in my case it is zsh, however, I'm using bash, hence foundry's path coudn't be found.
If that's the case, by simply using zsh or the default shell instead should solve the problem. 

However, if you plan on using other shells like bash, you'll need to add the config to to your selected shell's config file. 
Take bash for example, first find your bash config file, like /Users/<username>/.bashrc
Then add this line `export PATH="$PATH:/Users/<username>/.foundry/bin"` to it.
That tells bash there's this foundry thing you need to include. 
Then save the file, run `source /Users/<username>/.bashrc` to load the config file. 
After that it should be working.

Make sure your shell's config file path is correct, if it's wrong, you'll have to edit /Users/<username>/.bash_profile and change the path of .bashrc to the .bashrc we just edited.

**Reply by EngrPips** [reply]
Interesting, Thanks for this detailed heads-up
